File: Java/Boat.java
import java.util.Map;
import java.util.HashMap;
import java.time.DayOfWeek;

public class Boat extends Transportation {
    public Boat() {
        super(new HashMap<DayOfWeek, Integer>() {{
            put(DayOfWeek.MONDAY, 40);
            put(DayOfWeek.TUESDAY, 40);
            put(DayOfWeek.WEDNESDAY, 0);
            put(DayOfWeek.THURSDAY, 40);
            put(DayOfWeek.FRIDAY, 40);
            put(DayOfWeek.SATURDAY, 0);
            put(DayOfWeek.SUNDAY, 0);
        }});
    }
}

File: C#/KnowItTest/Transportations/Boat.cs
public class Boat : Transportation
{
    public Boat() : base(new Dictionary<DayOfWeek, int>
    {
        { DayOfWeek.Monday, 40 },
        { DayOfWeek.Tuesday, 40 },
        { DayOfWeek.Wednesday, 0 },
        { DayOfWeek.Thursday, 40 },
        { DayOfWeek.Friday, 40 },
        { DayOfWeek.Saturday, 0 },
        { DayOfWeek.Sunday, 0 }
    })
    {
        // Constructor body 
    }
}